src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -1,455 +1,702 @@
+                pw.write("GET http://" + thirdpartyHost + ":" + thirdpartyPort + "/ HTTP/1.1\n");
+                pw.write("Host: " + thirdpartyHost + "\n\n");
+                pw.flush();
+
+                // read the response via a 4k buffer
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                byte[] buffer = new byte[1024 * 4];
+                int bytesRead = is.read(buffer);
+                int totalBytesRead = 0;
+                while (bytesRead > -1) {
+                    totalBytesRead += bytesRead;
+                    bos.write(buffer, 0, bytesRead);
+                    bytesRead = is.read(buffer);
+                }
+                String response = new String(bos.toByteArray());
+                if (log.isDebugEnabled())
+                    log.debug("Response is " + totalBytesRead + " bytes: \n" + response);
+
+                Matcher m = thirdPartyContentPattern.matcher(response);
+                if (m.matches()) {
+                    log.debug("Response matches expected third party pattern!");
+                    is.close();
+                    os.close();
+                    bos.close();
+                    socket.close();
+                    return true;
+                } else {
+                    log.debug("Response does *not* match expected third party pattern");
+                }
+                is.close();
+                os.close();
+                bos.close();
+                socket.close();
+                return false;
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
+                    log.debug("The status line returned: " + statusLine);
+                }
+                return false;
+            }
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
+            }
+        }
+        return false;
+    }
+
+    private final MessageAndEvidence testHttpMethod(String httpMethod) {
+        //httpMethod = httpMethod.toLowerCase().trim();
+
+        //TODO:chose a body that exploits a vulnerability
+        String randomKey = RandomStringUtils.random(15, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+        String randomValue = RandomStringUtils.random(15, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789");
+
+        String requestBody = '"'+randomKey+"\":\""+randomValue+'"'; //"randomKey":"randomValue"
+        if (httpMethod.equals(HttpRequestHeader.TRACK) || httpMethod.equals(HttpRequestHeader.TRACE) || httpMethod.equals(HttpRequestHeader.CONNECT)) {
+            /* We have separate methods for them */
+            return null;
+        }

```

Comment:
Hopefully this isn't really an issue since it's all local?
If this really needs to stay it may as well move before the random assignments.